sdk/node/src/connection.js

```diff
@@ -66,11 +66,13 @@ class Connection {
    *
    * @param {String} baseUrl Chain Core URL.
    * @param {String} token   Chain Core client token for API access.
+   * @param {String} agent   https.Agent used to provide TLS config.
    * @returns {Client}
    */
-  constructor(baseUrl, token = '') {
+  constructor(baseUrl, token = '', agent = {}) {
     this.baseUrl = baseUrl
     this.token = token || ''
+    this.agent = agent || {}
```

Comment:
I think if you do `this.agent = agent` with no defaulting, you can avoid checking for object key count in the request function below.